tests/unit_tests/test_oedisi_tools/broker/server.py

```diff
@@ -0,0 +1,86 @@
+from fastapi import FastAPI, BackgroundTasks, UploadFile
+from fastapi.exceptions import HTTPException
+from fastapi.responses import FileResponse
+
+from pathlib import Path
+
+import helics as h
+import grequests
+import traceback
+import uvicorn
+import logging
+import socket
+import time
+import yaml
+import sys
+import os
+
+BASE_PATH = Path(__file__).parent
+os.chdir(BASE_PATH)
+app = FastAPI()
+
+    
+def read_settings():
+    component_map = {}
+    yaml_file = BASE_PATH / "docker-compose.yml"
+    assert yaml_file.exists(), f"{yaml_file} does not exist"
+    with open(yaml_file, "r") as stream:
+        config = yaml.safe_load(stream)
+    services = config['services']
+    print(services)
+    broker = services.pop('oedisi_broker')
+    broker_ip = broker['networks']['custom-network']['ipv4_address']
+    api_port = int(broker['ports'][0].split(":")[0])
+
+    for service in services:
+        ip = services[service]['networks']['custom-network']['ipv4_address']
+        port = int(services[service]['ports'][0].split(":")[0])
+        component_map[ip] = port
+        
+    return  services, component_map, broker_ip, api_port
+
+
+         
+
```

Comment:
Nit: Maybe run a formatter? There's a lot of space here.

Reply:
ran black